packages/compile-solidity/src/run.ts

```diff
@@ -216,7 +279,10 @@ function prepareOutputSelection({ targets = [] }) {
 /**
  * Load solc and perform compilation
  */
-async function invokeCompiler({ compilerInput, options, solc }) {
+async function invokeCompiler({ compilerInput, options, solc }): Promise<{
+  compilerOutput: CompilerOutput;
+  solcVersion: string;
+}> {
```

Comment:
How does this compile without a type definition for the args? I think you may have a bunch of things that are implicitly typed as `any`.

I'd recommend turning on the `noImplicitAny` flag in `tsconfig.json` and seeing what pops up.

Reply:
FWIW, since this code bumps up against a bunch of stuff that's untyped, please don't interpret my suggestion here as trying to require you to enable `noExplicitAny` in order to merge this. I do think this code would probably be improved if `noImplicitAny` were `true`, but I was more just suggesting temporarily enabling it to spot any implicit `any` types to make it easier to decide whether it's possible/appropriate to make them explicitly typed.

Net net, I think this PR is already well over the threshold of value added to our code, so I'm likely to approve once you're done pushing changes.